### PR TITLE
Replace HTTP URLs with HTTPS for sites with HTTPS versions

### PR DESCRIPTION
**Pull request by aaronfranke**
This PR replaces HTTP URLs with HTTPS where applicable (only for sites with HTTPS versions, and also example/dummy links). I also fixed a few instances of outdated links, such as `http://continuousphysics.com/Bullet` -> `https://bulletphysics.org`.

While making this PR I noticed that there are many URLs that are dead links. I left all of those alone, but at some point another PR should replace those with working links or with web.archive.org links.

### CODE REVIEW

File: COPYRIGHT.txt
Comment (Faless) [inline COPYRIGHT.txt:488]: We should not change the "copying" files. That's the suggested url in the official page:
https://www.apache.org/licenses/LICENSE-2.0#apply

File: COPYRIGHT.txt
Comment (Faless) [inline COPYRIGHT.txt:1138]: this seem to have been updated to https in the version 3, so I think this is fine?

File: COPYRIGHT.txt
Comment (neikeq) [inline COPYRIGHT.txt:488]: Are you sure it's fine to edit the urls in the text of the license?

File: COPYRIGHT.txt
Comment (Faless) [inline COPYRIGHT.txt:1352]: Are we using an old version of the FTL?
https://www.freetype.org/license.html
The last update seems `2006-Jan-27`, while we have `2000-Feb-08`. Should be double checked

**Review by Faless — COMMENTED** [review]
See notes on licences' changes.

File: COPYRIGHT.txt
Comment (pycbouh) [reply COPYRIGHT.txt:488]: For the reference, they still use HTTP on the same page that this links to. 🙃 And they don't redirect HTTP to HTTPS, for some reason.

https://www.apache.org/licenses/LICENSE-2.0#apply

Comment (aaronfranke) [reply COPYRIGHT.txt:488]: I have a feeling that we should have Apache fix it on their end.

Comment (aaronfranke) [reply COPYRIGHT.txt:488]: Their website actually contains a mix of HTTP and HTTPS links, this one is HTTPS:

apache

**Comment by Calinou** [comment]
I wouldn't edit any third-party license notice, since we don't control them.

**Comment by aaronfranke** [comment]
I undid the changes to the Apache licenses and FreeType. I also sent a message to the Apache maintainers about this. For other licenses, the GPL and MPL both use HTTPS links in the official version of the licenses and auto-redirect to HTTPS, so I kept them.

**Review by mhilbrunner — APPROVED** [review]

**Comment by mhilbrunner** [comment]
Thanks!

File: COPYRIGHT.txt
Comment (akien-mga) [reply COPYRIGHT.txt:1138]: No this was wrong. This is the glslang license, and it uses http. We should *not* modify thirdparty project license terms.

You can PR the change to https://github.com/KhronosGroup/glslang/blob/master/LICENSE.txt#L321 but until approved upstream, we shouldn't change their license.

File: core/math/convex_hull.h
Comment (akien-mga) [inline core/math/convex_hull.h:52]: This was bogus, the https link gives error 502. The http link worked.

File: editor/translations/el.po
Comment (akien-mga) [inline editor/translations/el.po:9]: This is also wrong, that's the identity that @OverloadedOrama uses on Weblate, we shoudn't be modifying it.

File: core/math/convex_hull.h
Comment (pycbouh) [reply core/math/convex_hull.h:52]: HTTP link redirects to https://code.google.com/archive/p/bullet/issues/275, so that's what we should've used.

File: servers/physics_3d/gjk_epa.cpp
Comment (akien-mga) [inline servers/physics_3d/gjk_epa.cpp:40]: We do not tamper with copyright statements.

File: core/math/convex_hull.h
Comment (aaronfranke) [reply core/math/convex_hull.h:52]: My bad, I thought I tested them all but I guess I missed this one.

File: servers/physics_3d/gjk_epa.cpp
Comment (aaronfranke) [reply servers/physics_3d/gjk_epa.cpp:40]: The old URL is completely dead. It leads to a WordPress page with this on it:

เว็บดูหนังออนไลน์ฟรี - หนัง หนังไทย ดูการ์ตูนสุดมันเพลิดเพลินใจ

Comment (akien-mga) [reply servers/physics_3d/gjk_epa.cpp:40]: Yeah the URL is bogus, but that's not our role to update copyright statements or licenses. Those are legal aspects and we should respect what was included in the code we vendored, even if it's bogus.

For example the [CC-BY 3.0](https://creativecommons.org/licenses/by/3.0/legalcode) license has a typo in "MERCHANTIBILITY", and I've made sure to prevent `codespell` from fixing it as it's not our job to do so (and we're modifying a standard legal document if we do). Now it's been fixed in CC-BY 4.0 so 6d26ff4eda3e37f19653cf0f04d10790475cc2c1 solved it, but if we add back CC-BY 3.0 content, we'll have to respect this typo :)

> A better solution would be to ask Bullet to update the URL.

That's worth doing since they have bogus URLs in their current code. It's still debatable whether we should also update it in the 10 year old version of their code we vendor, or if we should respect whatever was their copyright statement at the time.

File: editor/translations/el.po
Comment (aaronfranke) [reply editor/translations/el.po:9]: For the record Orama did reply to me via email that it's fine to use HTTPS for them if we want to.

Comment (OverloadedOrama) [reply editor/translations/el.po:9]: Oops, I missed the tag. Feel free to remove that line completely, I have changed my Weblate nickname and removed the website URL completely, and now I appear in el.po twice 😅

**Pull request merged**
